digdag-core/src/main/java/io/digdag/core/schedule/SchedulerManager.java

```diff
@@ -82,10 +94,47 @@ private Scheduler getScheduler(Config schedulerConfig, ZoneId workflowTimeZone)
             c.set("_command", command);
         }
 
+        for(String param : BUILT_IN_SCHEDULE_PARAMS){
+            if (param.equals("skip_on_overtime"))
+                c.get("skip_on_overtime", boolean.class, false);
+            else if (param.equals("skip_delayed_by"))
+                c.getOptional("skip_delayed_by", DurationParam.class);
+        }
+
         SchedulerFactory factory = types.get(type);
         if (factory == null) {
             throw new ConfigException("Unknown scheduler type: " + type);
         }
+
+        if (!usedKeys.isAllUsed()) {
+            shouldBeUsedKeys.removeAll(usedKeys);
+            if (!shouldBeUsedKeys.isEmpty()) {
+                warnUnusedKeys(shouldBeUsedKeys, usedKeys);
+            }
+        }
+
         return factory.newScheduler(c, workflowTimeZone);
     }
+
+    private void warnUnusedKeys(Set<String> shouldBeUsedButNotUsedKeys, Collection<String> candidateKeys)
+    {
+        // throw for only first unused key
+        for (String key : shouldBeUsedButNotUsedKeys) {
+            StringBuilder buf = new StringBuilder();
+            buf.append("Parameter '");
+            buf.append(key);
+            buf.append("' is not used at schedule. ");
+
+            List<String> suggestions = EditDistance.suggest(key, candidateKeys, 0.50);
+            if (suggestions.isEmpty()) {
+                throw new ConfigException(buf.toString());
+            }
+            else {
+                buf.append("> Did you mean '");
+                buf.append(suggestions);
+                buf.append("'");
+                throw new ConfigException(buf.toString());
```

Comment:
Throwing exception here is dangerous because:

* Schedules already saved in DB might already have unused keys. Those schedules are currently running in production.
* If here throws an exception, such schedules will stop working (ScheduleExecutor.runSchedule is calling SchedulerManager.getScheduler).
* From user's point of view, they can't notice that their schedules have stopped (because there are no notifications). It's unexpected from the users.
* From our point of view, ScheduleExecutor.runSchedule starts logging exceptions. It's not UncaughtException, but almost.

So, unused keys checking must be done only when a new workflow is uploaded. And that's good enough.

Reply:
To make it possible, unused keys checking in this getScheduler method needs to be optional.
A possible way is adding `boolean checkUnusedKeys` argument to this getScheduler method. Alternative way is adding `Optional<Consumer<Map<String, List<String>>>> unusedKeysCallback` argument to this getScheduler method (do the unused keys checking if unusedKeysCallback.isPresent()).
Either way works. I think either is fine.

Reply:
Another way I came up right now is adding a class like `UnusedConfigExceptions extends ConfigException`.
...either way works.

Reply:
I fixed to use unusedKeysCallback and UnusedConfigExceptions, but it seems like same as `boolean checkUnusedKeys`, so sorry if I'm misunderstanding your intension.
https://github.com/treasure-data/digdag/pull/1381/commits/db6f9434eea7980ff8fafdc3da1f04ac5359ab3b